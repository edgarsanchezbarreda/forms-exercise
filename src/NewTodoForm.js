import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

export const NewTodoForm = ({ addTodo }) => {
    const [task, setTask] = useState('');

    const handleChange = evt => {
        setTask(evt.target.value);
    };

    const handleSubmit = evt => {
        evt.preventDefault();
        addTodo({ task, id: uuidv4() });
        setTask('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor='task'>New Task</label>
            <input
                type='text'
                name='task'
                id='task'
                placeholder='todo'
                value={task}
                onChange={handleChange}
            />
            <button>Add Todo</button>
        </form>
    );
};
