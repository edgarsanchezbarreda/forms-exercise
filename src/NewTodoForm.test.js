import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { NewTodoForm } from './NewTodoForm';
import { add } from 'nodemon/lib/rules';

it('Renders without crashing', () => {
    render(<NewTodoForm />);
});

it('Matches snapshot', () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it('Runs addTodo function', () => {
    const addTodoMock = jest.fn();
    const { getByText } = render(<NewTodoForm addTodo={addTodoMock} />);
    const addButton = getByText('Add Todo');
    fireEvent.click(addButton);
    expect(addTodoMock).toHaveBeenCalled();
});
