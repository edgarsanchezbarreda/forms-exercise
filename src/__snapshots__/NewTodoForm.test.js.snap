// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      New Task
    </label>
    <input
      id="task"
      name="task"
      placeholder="todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
